scripts/gen_gdt.py

```diff
@@ -123,7 +123,10 @@ def parse_args():
             help="Print extra debugging information")
     parser.add_argument("-o", "--output-gdt", required=True,
             help="output GDT binary")
-    args = parser.parse_args()
+    args = parser.parse_args(
+        sys.argv[1:] +
+        (['--verbose'] if os.environ.get('VERBOSE') else [])
```

Comment:
I'd find doing this kinda hacky. Why not just set args.verbose after call to parse_args() ?

Reply:
I considered that, but I don't feel that would be less hacky, I would prefer to treat args as immutable.

Reply:
Personal preferences aside, I'm not sure in what such an idea would be grounded. [parse_args()](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args) takes an optional [Namespace](https://docs.python.org/3/library/argparse.html#argparse.Namespace) param, and can fill it in (if not given a new instance will be created and filled).

Additional quote not leading to ideas of immutability:

> It may also be useful to have an ArgumentParser assign attributes to an already existing object, rather than a new Namespace object. This can be achieved by specifying the namespace= keyword argument:

Finally, what the overall idea of this Namespace object? The idea is that it contains options, which may come from different sources: from command line args (handled by argparse.parse_args()), from a config file (of your favorite format, parsed by you), etc.

Reply:
Fair enough, will fix. Was not aware of the Namespace param.

Reply:
How about this instead? https://stackoverflow.com/a/10551389

Reply:
Though note that  os.environ.get('VERBOSE') is enough. Adding ", None" to the get call is redundant, as None is the default return value if the key is missing.

Reply:
> How about this instead? https://stackoverflow.com/a/10551389

I summarized my concern with defaults-which-aren't-really-defaults in the comment to that SO answer: https://stackoverflow.com/questions/10551117/setting-options-from-environment-variables-when-using-argparse/10551389#comment83184791_10551389 . It's still better than patching argv to pass to parse_args() though ;-).